File: be/app/Models/LearningPath.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class LearningPath extends Model
{
    protected $fillable = [
        'title',
        'description',
        'steps', // json array of step descriptors
    ];

    protected $casts = [
        'steps' => 'array',
    ];
}

File: fe/lib/models/learning_path.dart
class LearningPath {
  final int id;
  final String title;
  final String? description;
  final List<dynamic>? steps;

  LearningPath({required this.id, required this.title, this.description, this.steps});

  factory LearningPath.fromJson(Map<String, dynamic> json) {
    return LearningPath(
      id: json['id'],
      title: json['title'] ?? '',
      description: json['description'],
      steps: json['steps'] != null ? List.from(json['steps']) : null,
    );
  }
}
